src/StatusQ/Components/StatusListItem.qml

```diff
@@ -115,10 +133,17 @@ Rectangle {
                 text: statusListItem.title
                 font.pixelSize: 15
                 color: {
-                  if (statusListItem.type === StatusListItem.Type.Primary) {
-                      return Theme.palette.directColor1
+                  switch (statusListItem.type) {
+                      case StatusListItem.Type.Primary:
+                          return Theme.palette.directColor1
+                          break;
```

Comment:
You will get QML `Error : Cannot assign [undefined] to color` in case someone introduces a new Type and forget to update every switch/if in this component.
Because not every switch/if would be using (explicitly) the Danger enum for example, then a programmer won't be able to "find all usages" to find all places he/she needs to update while introducing a new Type.
This is a good illustration for our discussions we had about palettes BTW.

Reply:
Yes, very fair.